### PR TITLE
MSApiModule: better handling of empty responses & `ERR_AUTH_CODE_REFRESH`

### PR DESCRIPTION
**Pull request by dorschw**
when `empty_valid_codes` is passed, use it.

### CODE REVIEW

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 1]
```diff
@@ -177,9 +182,10 @@ def http_request(
         # 206 indicates Partial Content, reason will be in the warning header.
         # In that case, logs with the warning header will be written.
         if response.status_code == 206:
-            demisto.debug(str(response.headers))
-        is_response_empty_and_successful = (response.status_code == 204)
-        if is_response_empty_and_successful and return_empty_response:
+            demisto.debug(f'Response status code 206 (partial content) with headers {str(response.headers)}')
+
+        allowed_empty_response_codes = kwargs.get('empty_valid_codes', (204,))  # 204 is CommonServerPython's default
```
Comment (DeanArbel) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:187]: I don't understand, what the meaning of `CommonServerPython's default`?

File: Packs/AzureActiveDirectory/ReleaseNotes/1_2_11.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Azure Active Directory Identity Protection  (Deprecated)
+- Improved implementation of empty response handling in MicrosoftApiModule.
```
Comment (DeanArbel) [inline Packs/AzureActiveDirectory/ReleaseNotes/1_2_11.md:4]: `MicrosoftApiModule` makes little sense to the average user.
Please make sure to add it to all relevant files.
```suggestion
- Improved handling of empty responses.
```

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py [hunk 3]
```diff
@@ -50,6 +50,11 @@ class Scopes:
     'https://microsoftgraph.chinacloudapi.cn': 'cn'
 }
 
+# possible errors
+AUTH_CODE_REFRESH_FAIL = "Unable to create refresh_token with the provided authorization code. Please make " \
```
Comment (DeanArbel) [inline Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:54]: On a 2nd thought, let's rename this to `ERR_AUTH_CODE_REFRESH`

**Review by DeanArbel — APPROVED** [review]
Nice.

File: Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py (hunk 1 above)
Comment (dorschw) [reply Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule.py:187]: https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L8531-L8533

**Pull request closed without merging**